proposals/namespace_editing/README.md

```diff
@@ -0,0 +1,480 @@
+
+The UsdNamespaceEditor class will provide the following editing operations that work on object paths:
+
+* Delete[Prim/Property]AtPath(stage, path)
+	+ Authors/deletes **all scene description in the layer stack of the current edit target necessary** to remove the prim or property at path from the composed stage.
+
+* Move[Prim/Property]AtPath(stage, currentPath, newPath)
+	+ Authors/deletes all scene description in the layer stack of the current edit target necessary to make the composed prim or property at the currentPath instead exist at newPath.
+
+Additionally, the class will provide operations that take UsdPrim or UsdProperty objects for additional convenience. For each of the following XXXPrim functions that take a UsdPrim, there will be an equivalent XXXPropertry function that takes a UsdProperty
+
+* DeletePrim(usdPrim)
+	+ Authors/deletes all scene description in the layer stack of the current edit target necessary to remove the usdPrim from its composed stage.
+
+* RenamePrim(usdPrim, newName)
+	+ Authors all scene description in the layer stack of the current edit target necessary to change the name of the composed usdPrim to newName. The prim will retain its original parent prim.
+
+* ReparentPrim(usdPrim, newPrimParent)
+	+ Authors all scene description in the layer stack of the current edit target necessary to move the composed usdPrim from its original path to instead be a child of the composed prim, newPrimParent.
+
+* ReparentPrim(usdPrim, newPrimParent, newName)
+	+ Overload of ReparentPrim that also renames the composed prim to newName when moving it be a child of newParentPrim.
+
+For each of the above functions on both paths and UsdObjects, we will provide a corresponding Can[Operation] function (e.g. CanDeletePrimAtPath, CanMovePrim, CanRenameProperty) that returns whether the operation can be successfully performed with the current edit settings (see Authoring Settings below).
+
+#### Change Notifications
+
+Because namespace operations performed through the UsdNamespaceEditor API have a clear intent, we will be able to add the information about whether a UsdObject has been deleted, moved or renamed to UsdNotice::ObjectsChanged when edits are made through this API. In addition to the existing ObjectsChanged queries of `GetResyncedPaths()` and `GetChangedInfoOnlyPaths()`, we will add queries along the lines of `GetRenamedPaths()`, `GetMovedPaths()`, and `GetDeletedPaths()` to return what objects have been specifically edit using the UsdNamespaceEditor. We'll also provide the equivalent functions of `RenamedObject(UsdObject)`, `MovedObject(UsdObject)`, and `DeletedObject(UsdObject)` for querying a specific object as to whether it was namespace edited. Clients can make use of this new notice information to respond to changes a more efficient manner for namespace changes they know how to handle.
```

Comment:
> Clients can make use of this new notice information to respond to changes a more efficient manner for namespace changes they know how to handle.

Does this imply that we will still be including ResyncedPaths for all moved/renamed objects in the same Notice, as well as info/resynced paths for all the paths that needed to be "fixed up"?  If so, it seems tricky for clients that **do** handle the new, smart changes to avoid doing all the expensive updates.. but if **not**, it seems like clients who are not in-the-know would be completely missing out?

Reply:
Yes, I intend that ResyncedPaths will include all the moved/renamed objects precisely because it's still correct that these paths have been resynced. And the existence of downstream clients that are already looking at ResyncedPaths is an additional compelling reason as to why we'd want to keep it this way.
While it may be trickier for clients that can optimize based on rename/move to filter those out, the provided functions should be able to provide that filtering. We could additionally add an optional filter to GetResyncedPaths that would allow clients to filter out any combination of namespace edited paths if that proves to be useful.

Reply:
Thanks, @crevilla ! I reckon we can do this unambiguously for now … where it might get tricky is when we have general batch editing in usd, and then distinguishing the two cases “moved prim a and changed a property on it” from “moved prim a and switched a variant on it”.  
Both would have a resync for a, but in only one case can it be “ignored”